### ISSUE TITLE
Set context

### ISSUE THREAD

**Issue by MickL** [dbfannin/ngx-logger#219]
Is it possible to set the context?

I am using the logger in a library which outputs: 

> 2021-02-05T12:58:01.192Z DEBUG [my-module-my-module.js:20803] Hello world

But my-module is actually the client that uses the library not the module from the library and not the service I used the logger within. 

I would like what is set what stands in the brackets. E.g. in [Nest.js](https://docs.nestjs.com/techniques/logger#using-the-logger-for-application-logging) you would do it like this:

```ts
import { Logger } from 'ngx-logger';

@Injectable()
class MyService {
  private logger = new Logger(MyService.name);
  
  doSomething() {
    this.logger.log('Doing something...');
  }
}
```

Or alternatively it could be done like this:

```ts
import { Logger } from 'ngx-logger';

@Injectable()
class MyService {
  constructor(private logger: Logger) {
     logger.setContext(MyService.name);
  }
  
  doSomething() {
    this.logger.log('Doing something...');
  }
}
```

**Comment by qortex** [comment]
With the recently released features, you can achieve that by doing:
- `disableFileDetails: true` in the logger. That removes the file information
- use a custom logger that proxies the logger call, and transforms it like
```typescript
return `[${this.context}] ${message}`;
```

Would that work for you?

My opinion is that it's preferrable to do it this way because it brings lots of flexibility to the user, rather than bake a specific callback into the library which would make the config more complicated (and then, why use brackets, how to nest contexts, etc.). But I'm open to hear good reasons to change my mind :)

**Comment by bmtheo** [comment]
Hi @MickL,
I just published v5 that allows you to customise any behavior of the logger and that would be enough for your feature
The doc is here : 
https://github.com/dbfannin/ngx-logger/blob/master/docs/customising.md
In your case you would need to customise the metadata.service.ts

Still your feature request is probably something I'll look into

**Comment by MRisto** [comment]
Would also be intrested in something like this. nestjs-pino has the option to setContext for the logger in a service. I find that really handy.

**Comment by JanPhill** [comment]
Hi @bmtheo,
could you provide an example on how to set a context within a component using the metadata.service.ts?

I also really like the idea of being able to set a context using a method like `logger.setContext()` as @MickL suggested.

**Comment by bmtheo** [comment]
This is not exactly what was asked, but it is manageable that way :

You need to overwrite writer.service.ts and then provide it in the module (see more info about this in [customising doc](https://github.com/dbfannin/ngx-logger/blob/master/docs/customising.md))

Code would look like this (untested) :
```
@Injectable()
export class ContextNGXLoggerWriterService extends NGXLoggerWriterService{

  protected override prepareMetaString(metadata: INGXLoggerMetadata, config: INGXLoggerConfig): string {
    const baseMetaString = super.prepareMetaString(metadata, config);

    const contextData = ''; // Get your context here (requires to code your own context management)

    return baseMetaString + ' ' + contextData;
  }

}

```

**Comment by bmtheo** [comment]
Feature is available in 5.0.11
See doc here https://github.com/dbfannin/ngx-logger/blob/master/docs/features.md and full example here https://github.com/dbfannin/ngx-logger/tree/master/projects/demo/src/app/custom-instance
